pkg/estimator/client/general.go

```diff
@@ -84,25 +175,55 @@ func (ge *GeneralEstimator) maxAvailableReplicas(cluster *clusterv1alpha1.Cluste
 		}
 	}
 
-	return int32(maximumReplicas)
+	return maximumReplicas
 }
 
-func getAllowedPodNumber(resourceSummary *clusterv1alpha1.ResourceSummary) int64 {
-	var allocatable, allocated, allocating int64
-	if resourceSummary.Allocatable != nil {
-		allocatable = resourceSummary.Allocatable.Pods().Value()
-	}
-	if resourceSummary.Allocated != nil {
-		allocated = resourceSummary.Allocated.Pods().Value()
-	}
-	if resourceSummary.Allocating != nil {
-		allocating = resourceSummary.Allocating.Pods().Value()
+func getMaximumReplicasBasedOnResourceModels(cluster *clusterv1alpha1.Cluster, replicaRequirements *workv1alpha2.ReplicaRequirements) (int64, error) {
+	resourceModelsMinMap := convertToResourceModelsMinMap(cluster.Spec.ResourceModels)
+
+	minCompliantModelIndex := 0
+	for key, value := range replicaRequirements.ResourceRequest {
+		requestedQuantity := value.Value()
+		if requestedQuantity <= 0 {
+			continue
+		}
+
+		quantityArray, ok := resourceModelsMinMap[clusterv1alpha1.ResourceName(key)]
+		if !ok {
+			return -1, fmt.Errorf("we cannot use resource models to calculate max available replicas due to the resource is not in it")
+		}
+
+		for index, minValue := range quantityArray {
+			// Suppose there is the following resource model:
+			// Model1: cpu [1C,2C)
+			// Model2: cpu [2C,3C)
+			// if pod cpu request is 1.5C, we regard the nodes in model1 as meeting the requirements of the Pod.
+			// Suppose there is the following resource model:
+			// Model1: cpu [1C,2C), memory [1Gi,2Gi)
+			// Model2: cpu [2C,3C), memory [2Gi,3Gi)
+			// if pod cpu request is 1.5C and memory request is 2.5Gi
+			// We regard the node of model1 as not meeting the requirements, and the nodes of model2 and later as meeting the requirements.
+			if minValue.Cmp(value) > 0 {
+				// if index == 0, minValue.Cmp(value) must <= 0
```

Comment:
```suggestion
				// Since the 'min' value of the first model is always 0, hit here
				// the index should be >=1, so it's safe to use 'index-1' here.
```